docs/migrating-v1alpha1-to-v1beta1.md

```diff
@@ -0,0 +1,189 @@
+      type: string
+
+# TaskRun.yaml (v1alpha1)
+spec:
+  inputs:
+    params:
+    - name: ADDR
+      value: https://example.com/foo.json
+```
+
+Equivalent v1beta1 Params:
+
+```yaml
+# Task.yaml (v1beta1)
+spec:
+  params:
+  - name: ADDR
+    description: Address to curl.
+    type: string
+
+# TaskRun.yaml (v1beta1)
+spec:
+  params:
+  - name: ADDR
+    value: https://example.com/foo.json
+```
+
+#### Input and Output PipelineResources
+
+In v1alpha1, `PipelineResources` were stored under `spec.inputs.resources` and
+`spec.outputs.resources`. In v1beta1 these change to `spec.resources.inputs` and
+`spec.resources.outputs`.
+
+Example v1alpha1 PipelineResources:
+
+```yaml
+# Task.yaml (v1alpha1)
+spec:
+  inputs:
+    resources:
+    - name: skaffold
+      type: git
+  outputs:
+    resources:
+    - name: baked-image
+      type: image
+
+# TaskRun.yaml (v1alpha1)
+spec:
+  inputs:
+    resources:
+    - name: skaffold
+      resourceSpec:
+        type: git
+        params:
+          - name: revision
+            value: v0.32.0
+          - name: url
+            value: https://github.com/GoogleContainerTools/skaffold
+  outputs:
+    resources:
+    - name: baked-image
+      resourceSpec:
+      - type: image
+        params:
+        - name: url
+          value: gcr.io/foo/bar
+```
+
+Equivalent v1beta1 PipelineResources:
+
+```yaml
+# Task.yaml (v1beta1)
+spec:
+  resources:
+    inputs:
+    - name: src-repo
+      type: git
+    outputs:
+    - name: baked-image
+      type: image
+
+# TaskRun.yaml (v1beta1)
+spec:
+  resources:
+    inputs:
+    - name: src-repo
+      resourceSpec:
+        type: git
+        params:
+          - name: revision
+            value: master
+          - name: url
+            value: https://github.com/tektoncd/pipeline
+    outputs:
+    - name: baked-image
+      resourceSpec:
+      - type: image
+        params:
+        - name: url
+          value: gcr.io/foo/bar
+```
+
+## PipelineResources and Catalog Tasks
+
+PipelineResources are remaining in alpha while the other resource kinds are
+promoted to beta. This is part of a deliberate effort to migrate users away
+from PipelineResources onto a broader suite of reusable Tasks provided by
+the Tekton catalog.
+
+In the near-term `PipelineResources` will continue to be supported by Tekton.
+At some point the feature may well be deprecated and subsequently replaced.
+
+To ease migration away from PipelineResources each type will soon have
+an equivalent Task in the Catalog. Alongside the Task will be documentation
+showing how to migrate a Pipeline over to it.
+
+### Git Resource
+
+The [`git-clone` Task in the Catalog](https://github.com/tektoncd/catalog/tree/v1beta1-wip/git)
+provides an equivalent to the Git Resource.
+
+### Pull Request Resource
+
+The [`pullrequest` Task in the Catalog](https://github.com/tektoncd/catalog/tree/v1beta1-wip/pullrequest)
+provides an equivalent to the PullRequest Resource.
+
+### Image Resource
+
+WIP
```

Comment:
This one is trickier than the other. We can provide a task for it, but most likely it's way more efficient to have a step for the image builder that support this, like [here](https://github.com/tektoncd/catalog/pull/216/files#diff-b5ed113f189d0222f799ae586f1ecee5R44)

Reply:
Ahhh I see.  So instead of an explicit Task called "ImageOutput", this is really just replaced with a digest result / params or similar.  And any task that generates an image could receive / output those.  Makes sense.